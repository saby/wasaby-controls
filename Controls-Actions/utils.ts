import { IContextValue } from 'Controls/context';
import { Record } from 'Types/entity';

export interface IContextData extends IContextValue {
    documentContext?: Record;
}

/**
 * Получает объект из контекста
 * @public
 * @param context
 * @param objectName
 */
function findObject(context: IContextData, objectName: string): unknown {
    return context?.[objectName];
}

export { findObject };
