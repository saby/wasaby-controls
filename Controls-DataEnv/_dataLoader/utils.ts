import { loadAsync } from 'WasabyLoader/ModulesLoader';
import { addPageDeps } from 'UI/Deps';
import { logger as Logger } from 'Application/Env';
import { RPC } from 'Browser/Transport';

function resolveLogger(error: Error): typeof Logger.error | typeof Logger.warn {
    let logger = Logger.error;

    if (error instanceof RPC.Error) {
        logger = error.errType === 'warning' ? Logger.warn : Logger.error;
    }

    return logger;
}

export async function loadModule<TModule>(moduleName: string): Promise<TModule | undefined> {
    try {
        const module = await loadAsync<TModule>(moduleName);
        addPageDeps([moduleName]);
        return module;
    } catch (error: unknown) {
        loadErrorHandler(
            error,
            `Controls-DataEnv/dataLoader::Ошибка при загрузке модуля ${moduleName}`
        );
    }
}

export function loadErrorHandler(e: unknown, message: string): void {
    if (e instanceof Error) {
        const logger = resolveLogger(e);
        logger(`${message} ${e.message}`);
    }
}
