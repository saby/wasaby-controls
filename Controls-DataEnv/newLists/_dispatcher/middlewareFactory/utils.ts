export const addName = <T extends Function>(label: string, fn: T): T => {
    const wrapper = {
        [label]: ((...args: unknown[]) => fn(...args)) as unknown as T,
    };

    return wrapper[label];
};

const POSTFIX = '{name}';
export const createMask = <T extends string>(prefix: T) => `${prefix}(${POSTFIX})` as const;
export const decorateName = (mask: string, name: string) => mask.replace(POSTFIX, name);
export const undecorateName = (mask: string, decoratedName: string) => {
    const [start, end] = mask.split(POSTFIX);
    return decoratedName.replace(start, '').replace(end, '');
};
