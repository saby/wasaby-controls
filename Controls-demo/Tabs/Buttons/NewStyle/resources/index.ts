import { RecordSet } from 'Types/collection';

export const ITEMS = new RecordSet({
    keyProperty: 'id',
    rawData: [
        {
            id: '1',
            title: 'Document',
        },
        {
            id: '2',
            title: 'Files',
        },
        {
            id: '3',
            title: 'Orders',
        },
    ],
});

export const ITEMS_WITH_ICON_AND_COUNTER = new RecordSet({
    keyProperty: 'id',
    rawData: [
        {
            id: '1',
            title: 'Document',
        },
        {
            id: '2',
            title: 'Files',
            mainCounter: 3,
        },
        {
            id: '3',
            title: 'Orders',
            icon: 'icon-EmptyMessage',
            iconStyle: 'label',
        },
    ],
});
