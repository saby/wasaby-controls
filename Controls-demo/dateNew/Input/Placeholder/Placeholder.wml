<div class="controlsDemo__wrapper controlsDemo__ml3">
    <div class="controlsDemo__cell">
        <div class="controls-text-label"> placeholder = 'Когда?', calendarButtonVisible = false</div>
        <Controls.date:Input
                calendarButtonVisible="{{ false }}"
                placeholder="Когда?"
                _date="{{ _date }}"
                _displayDate="{{ _date }}"
                bind:value="_value2">
        </Controls.date:Input>
    </div>
    <div class="controlsDemo__cell">
        <div class="controls-text-label"> placeholder = 'Когда?', calendarButtonVisible = true</div>
        <Controls.date:Input
                calendarButtonVisible="{{ true }}"
                placeholder="Когда?"
                _date="{{ _date }}"
                _displayDate="{{ _date }}"
                bind:value="_value">
        </Controls.date:Input>
    </div>

    <div class="controlsDemo__cell">
        <div class="controls-text-label"> placeholder = 'Длинный placeholder', calendarButtonVisible = true</div>
        <Controls.date:Input
                calendarButtonVisible="{{ true }}"
                placeholder="Длинный placeholder"
                _date="{{ _date }}"
                _displayDate="{{ _date }}"
                bind:value="_value3">
        </Controls.date:Input>
    </div>
</div>
