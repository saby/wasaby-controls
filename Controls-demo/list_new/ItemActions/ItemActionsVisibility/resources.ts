import { IoC } from 'Env/Env';
import { Model } from 'Types/entity';

import { IItemAction, TItemActionShowType } from 'Controls/itemActions';

export interface ISrcData {
    key: number;
    title: string;
    description: string;
}

export const columns = [
    {
        displayProperty: 'key',
    },
    {
        displayProperty: 'title',
    },
    {
        displayProperty: 'description',
    },
];

export const srcData: ISrcData[] = [
    {
        key: 1,
        title: 'Notebooks 1',
        description: 'Другое название 1',
    },
    {
        key: 2,
        title: 'Notebooks 2',
        description: 'Описание вот такое',
    },
    {
        key: 3,
        title: 'Smartphones 3 ',
        description: 'Хватит страдать',
    },
    {
        key: 4,
        title: 'Notebooks 1',
        description: 'Другое название 1',
    },
    {
        key: 5,
        title: 'Notebooks 2',
        description: 'Описание вот такое',
    },
    {
        key: 6,
        title: 'Smartphones 3 ',
        description: 'Хватит страдать',
    },
];

export const menuItemActions: IItemAction[] = [
    {
        id: 1,
        icon: 'icon-PhoneNull',
        title: 'phone',
        handler(model: Model): void {
            IoC.resolve('ILogger').info('action phone Click ', model);
        },
    },
    {
        id: 2,
        icon: 'icon-EmptyMessage',
        title: 'message',
        parent: null,
        'parent@': true,
        handler(model: Model): void {
            alert('Message Click');
        },
    },
    {
        id: 3,
        icon: 'icon-Profile',
        title: 'profile',
        showType: TItemActionShowType.MENU,
        parent: 2,
        'parent@': null,
        handler(model: Model): void {
            IoC.resolve('ILogger').info('action profile Click');
        },
    },
    {
        id: 6,
        title: 'call',
        parent: 2,
        'parent@': null,
        handler(model: Model): void {
            IoC.resolve('ILogger').info('action profile Click');
        },
    },
];

export const menuToolbarItemActions: IItemAction[] = [
    {
        id: 5,
        title: 'прочитано',
        showType: TItemActionShowType.TOOLBAR,
        handler(model: Model): void {
            IoC.resolve('ILogger').info('action read Click');
        },
    },
    ...menuItemActions,
    {
        id: 4,
        icon: 'icon-Erase',
        iconStyle: 'danger',
        title: 'delete pls',
        showType: TItemActionShowType.TOOLBAR,
        handler(model: Model): void {
            IoC.resolve('ILogger').info('action delete Click');
        },
    },
];
