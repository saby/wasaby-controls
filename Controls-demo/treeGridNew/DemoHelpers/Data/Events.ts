import { IColumn } from 'Controls/grid';
import { IData } from '../Interface';

export const Events = {
    getColumns: (): IColumn[] => {
        return [
            {
                displayProperty: 'title',
                width: '',
            },
        ];
    },
    getDataBeforeExpanded: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Асинхронная обработка перед разворотом узла',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Асинхронная обработка перед разворотом узла',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataBeforeCollapsed: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Асинхронная обработка перед сворачиванием узла',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Асинхронная обработка перед сворачиванием узла',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataCollapsedItemsChanged: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Этот узел нельзя свернуть',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Этот узел можно свернуть',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataExpandedItemsChanged: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Этот узел нельзя развернуть',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Этот узел можно развернуть',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
};
