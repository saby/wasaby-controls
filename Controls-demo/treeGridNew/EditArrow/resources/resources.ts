import { IColumn, IHeaderCell } from 'Controls/grid';
import * as ColumnTemplate from 'wml!Controls-demo/treeGridNew/EditArrow/resources/ColumnTemplate';

export const TreeHeader: IHeaderCell[] = [
    {
        caption: '',
    },
    {
        caption: 'price',
        align: 'right',
    },
    {
        caption: 'count',
        align: 'right',
    },
];

export const TreeColumns: IColumn[] = [
    {
        displayProperty: 'title',
        width: '300px',
        textOverflow: 'ellipsis',
    },
    {
        displayProperty: 'price',
        width: '100px',
        align: 'right',
    },
    {
        displayProperty: 'count',
        width: '100px',
        align: 'right',
    },
];

export const TreeColumnsWithTemplate: IColumn[] = [
    {
        displayProperty: 'title',
        width: '300px',
        template: ColumnTemplate,
    },
    {
        displayProperty: 'price',
        width: '100px',
        align: 'right',
        template: ColumnTemplate,
    },
    {
        displayProperty: 'count',
        width: '100px',
        align: 'right',
        template: ColumnTemplate,
    },
];

export const TreeData = [
    {
        key: 1,
        parent: null,
        'parent@': true,
        title: 'First Node',
        description: 'description',
        price: '100',
        count: '10',
    },
    {
        key: 2,
        parent: null,
        'parent@': true,
        title: 'Second Node',
        description: 'description',
        price: '200',
        count: '30',
    },
    {
        key: 3,
        parent: 2,
        'parent@': true,
        title: 'Third Node with veeeery long caption, so it fits only in two lines',
        description: 'description',
        price: '100',
        count: '10',
    },
    {
        key: 4,
        parent: 3,
        'parent@': null,
        title: 'Fourth Node',
        description: 'description',
        price: '200',
        count: '30',
    },
    {
        key: 5,
        parent: null,
        'parent@': null,
        title: 'Leaf 1',
        description: 'description',
        price: '200',
        count: '30',
    },
    {
        key: 6,
        parent: 1,
        'parent@': null,
        title: 'Leaf 2',
        description: 'description',
        price: '200',
        count: '30',
    },
    {
        key: 7,
        parent: 2,
        'parent@': null,
        title: 'Leaf 3',
        description: 'description',
        price: '200',
        count: '30',
    },
];
