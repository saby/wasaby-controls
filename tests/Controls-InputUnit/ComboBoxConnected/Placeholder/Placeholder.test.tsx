/**
 * @jest-environment jsdom
 */

import { getContent } from 'Controls-InputUnit/resources/utils';
import { WasabyEvents } from 'UICore/Events';
import { unmountComponentAtNode } from 'react-dom';
import { render } from '@testing-library/react';
import { default as Placeholder } from 'Controls-Input-demo/ComboboxConnected/Placeholder';

describe('Controls-Input/ComboBoxGroupConnected/Placeholder', () => {
    setTestID(152174968);
    let container = null;

    beforeEach(() => {
        jest.useFakeTimers('modern');
        jest.setSystemTime(new Date(1996, 5, 21));
        container = document.createElement('div');
        WasabyEvents.initInstance(container);
        document.body.appendChild(container);
    });
    afterEach(() => {
        unmountComponentAtNode(container);
        WasabyEvents.destroyInstance(container);
        jest.useRealTimers();
        container.remove();
        container = null;
    });

    it('Placeholder', async () => {
        render(getContent(Placeholder), { container });
        expect(container).toMatchSnapshot();
    });
});
