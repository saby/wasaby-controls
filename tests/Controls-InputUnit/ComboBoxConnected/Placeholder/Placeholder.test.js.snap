// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls-Input/ComboBoxGroupConnected/Placeholder Placeholder 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo_fixedWidth300 tw-flex tw-flex-col"
  >
    <div
      class="tw-flex controls-max-w-full tw-flex-col controls-margin_top-m controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            With placeholder
          </div>
          
        </div>
      </a>
      <div
        class="controls-ComboBox controls_toggle_theme-undefined controls-ComboBox-borderStyle-valid   controls-ComboBox-borderStyle-valid_focusing controls-ComboBox_background-same controls-text-default controls-fontweight-default controls-inlineheight-default controls-Combobox-fontsize-m controls-ComboBox_textAlign-left controls-ComboBox_border-partial controls-margin_top-m"
        tabindex="0"
        title=""
      >
        <div
          class="controls-ComboBox-wrapper"
        >
          <div
            class="controls-ComboBox__field controls-ComboBox__field_margin-null "
            tabindex="0"
          >
            <div
              class="controls-ComboBox__placeholder  tw-cursor-text"
            >
              Кликните на вариант
            </div>
          </div>
          <span
            class="controls-ComboBox__arrow-wrapper controls-ComboBox__arrow-wrapper_offset-null"
          >
            <span
              class="controls-ComboBox__iconArrow icon-ExpandLight"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
